plugins/inputs/synproxy/synproxy_test.go

```diff
@@ -0,0 +1,146 @@
+// +build linux
+
+package synproxy
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/influxdata/telegraf/testutil"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSynproxyFileNormal(t *testing.T) {
+	tmpfile := makeFakeSynproxyFile([]byte(SynproxyFile_Normal))
+	defer os.Remove(tmpfile)
+
+	k := Synproxy{
+		statFile: tmpfile,
+	}
+
+	acc := testutil.Accumulator{}
+	err := k.Gather(&acc)
+	assert.NoError(t, err)
+
+	fields := map[string]interface{}{
+		"entries":        uint32(0x00000000),
+		"syn_received":   uint32(0x0003e27b),
+		"cookie_invalid": uint32(0x0001493e),
+		"cookie_valid":   uint32(0x0003d7cf),
+		"cookie_retrans": uint32(0x00000000),
+		"conn_reopened":  uint32(0x00000000),
+	}
+	acc.AssertContainsFields(t, "synproxy", fields)
+}
+
+func TestSynproxyFileOverflow(t *testing.T) {
+	tmpfile := makeFakeSynproxyFile([]byte(SynproxyFile_Overflow))
+	defer os.Remove(tmpfile)
+
+	k := Synproxy{
+		statFile: tmpfile,
+	}
+
+	acc := testutil.Accumulator{}
+	err := k.Gather(&acc)
+	assert.NoError(t, err)
+
+	fields := map[string]interface{}{
+		"entries":        uint32(0x00000000),
+		"syn_received":   uint32(0x00000004),
+		"cookie_invalid": uint32(0xd0000009),
+		"cookie_valid":   uint32(0x00000004),
+		"cookie_retrans": uint32(0x00000000),
+		"conn_reopened":  uint32(0x00000000),
+	}
+	acc.AssertContainsFields(t, "synproxy", fields)
+}
+
+func TestSynproxyFileHeaderMismatch(t *testing.T) {
+	tmpfile := makeFakeSynproxyFile([]byte(SynproxyFile_HeaderMismatch))
+	defer os.Remove(tmpfile)
+
+	k := Synproxy{
+		statFile: tmpfile,
+	}
+
+	acc := testutil.Accumulator{}
+	err := k.Gather(&acc)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "out of column boundary!")
+}
+
+func TestSynproxyFileInvalidHex(t *testing.T) {
+	tmpfile := makeFakeSynproxyFile([]byte(SynproxyFile_InvalidHex))
+	defer os.Remove(tmpfile)
+
+	k := Synproxy{
+		statFile: tmpfile,
+	}
+
+	acc := testutil.Accumulator{}
+	err := k.Gather(&acc)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "Invalid value")
+}
+
+func TestNoSynproxyFile(t *testing.T) {
+	tmpfile := makeFakeSynproxyFile([]byte(SynproxyFile_Normal))
+	os.Remove(tmpfile)
```

Comment:
defer os.Remove

Reply:
This function will test behavior when the expected file is missing. Therefore we remove the file before ending the test. I added comment in the source to make this more clear.